utensor_cgen/experimental/ugraph_builder.py

```diff
@@ -186,6 +189,214 @@ def QuantRangeForMultiplicationu8u8int32_Op(name, a_range, b_range, ugraph):
+                        input_tensors=inputs,
+                        output_tensors=[conv_out],
+                        op_type="Conv2D",
+                        backend="tensorflow",
+                        ugraph=tmp_ugraph)
+  
+  ugraph.add_op(conv2d_op_info, False)
+  return conv2d_op_info.output_tensors
+
+def quantized_conv2d_op(name, inputs, ugraph):
+  tmp_ugraph = uTensorGraph(output_nodes=[name])
+  conv_out = TensorInfo(name=name + ":0",
+                    op_name=name,
+                    dtype=np.dtype('uint8'),
+                    shape=inputs[0].shape, #FIXME: wrong shape most likely
+                    ugraph=tmp_ugraph
+                    )
+  min_out = TensorInfo(name=name + ":1",
+            op_name=name,
+            dtype=np.dtype('float32'),
+            shape=[1],
+            ugraph=tmp_ugraph
+            )
+  max_out = TensorInfo(name=name + ":2",
+              op_name=name,
+              dtype=np.dtype('float32'),
+              shape=[1],
+              ugraph=tmp_ugraph
+              )
+
+  quantized_conv2d_op_info = OperationInfo(name=name,
+                        input_tensors=inputs,
+                        output_tensors=[conv_out, min_out, max_out],
+                        op_type="QuantizedConv2D",
+                        backend="tensorflow",
+                        ugraph=tmp_ugraph)
+
+  ugraph.add_op(quantized_conv2d_op_info, False)
+  return quantized_conv2d_op_info.output_tensors
+
+def relu_op(name, inputs, ugraph):
+  tmp_ugraph = uTensorGraph(output_nodes=[name])
+  relu_out = TensorInfo(name=name + ":0",
+                    op_name=name,
+                    dtype=np.dtype('float32'),
+                    shape=inputs[0].shape,
+                    ugraph=tmp_ugraph
+                    )
+  relu_op_info = OperationInfo(name=name,
+                        input_tensors=inputs,
+                        output_tensors=[relu_out],
+                        op_type="Relu",
+                        backend="tensorflow",
+                        ugraph=tmp_ugraph)
+  
+  ugraph.add_op(relu_op_info, False)
+  return relu_op_info.output_tensors
+
+def maxpool_op(name, inputs, ugraph):
+  tmp_ugraph = uTensorGraph(output_nodes=[name])
+  max_out = TensorInfo(name=name + ":0",
+                    op_name=name,
+                    dtype=np.dtype('float32'),
+                    shape=inputs[0].shape,
+                    ugraph=tmp_ugraph
+                    )
+  max_op_info = OperationInfo(name=name,
+                        input_tensors=inputs,
+                        output_tensors=[max_out],
+                        op_type="MaxPool",
+                        backend="tensorflow",
+                        ugraph=tmp_ugraph)
+  ugraph.add_op(max_op_info, False)
+  return max_op_info.output_tensors
+
+def quantized_maxpool_op(name, inputs, ugraph):
+  tmp_ugraph = uTensorGraph(output_nodes=[name])
+  max_value_out = TensorInfo(name=name + ":0",
```

Comment:
It's not right to do this.
You are basically adding an op to `tmp_ugraph`.
I know why you have to write so many redundant helper functions.
Because you keep sharing ops among different `ugraph`s, even temp ones.